web/client-api/src/main/java/io/deephaven/web/client/api/barrage/BarrageUtils.java

```diff
@@ -364,6 +364,21 @@ private static ColumnData readArrowBuffer(ByteBuffer data, Iter<FieldNode> nodes
                 return new ShortArrayColumnData(Js.uncheckedCast(shortArray));
             case "boolean":
             case "java.lang.Boolean":
+                // noinspection IntegerDivisionInFloatingPointContext
+                assert positions.length().toFloat64() >= ((size + 63) / 64);
+                // booleans are stored as a bitset, but internally we represent booleans as bytes
+                data.position((int) positions.offset().toFloat64());
+                BitSet wireValues = readBitSetWithLength(data, (int) (positions.length().toFloat64()));
+                byte[] boolArray = new byte[size];
+                for (int i = 0; i < size; ++i) {
+                    if (!hasNulls || valid.get(i)) {
+                        boolArray[i] = (byte) (wireValues.get(i) ? 1 : 0);
```

Comment:
Optionally we can go further here and actually replace the underlying store to not go back to byte[] - there's no benefit either way. Downstream of this, there is more code that will read those byte[]s and return them to true/false/null.